zaga-web/src/app/components/creatework-order/creatework-order.component.html

```diff
@@ -1,857 +1,1708 @@
+                <td *ngIf="!new">
+                  {{ material.spent || 0 }} {{ material.material.unit }}
+                </td>
+                <td *ngIf="!new">
+                  {{ material.spent / workOrder.area }}
+                  {{ material.material.unit }}/ha
+                </td>
+                <td>
+                  <button
+                    class="btn btn-primary btn-edit mr-2"
+                    (click)="editMaterial(material)"
+                    [disabled]="workOrder.status == 'Zatvoren'"
+                  >
                     <i class="fa fa-edit"></i>
                   </button>
                 </td>
               </tr>
             </tbody>
           </table>
-
-        </div>
+        </form>
       </div>
     </div>
-
   </div>
-
 </div>
 
 <div class="container d-md-none">
-  <div class="row" style="margin-top: 10px;">
+  <div class="row" style="margin-top: 10px">
     <div class="col-8">
       <h3 *ngIf="new">Kreiranje naloga</h3>
       <h3 *ngIf="!new">Informacije o nalogu</h3>
-      <span>Status: </span><span
-        [ngClass]="{'statusNovi': workOrder.status == 'Novi', 'statusURadu': workOrder.status == 'U radu', 'statusZatvoren' : workOrder.status == 'Zatvoren'}">{{workOrder.status}}</span>
+      <span>Status: </span
+      ><span
+        [ngClass]="{
+          statusNovi: workOrder.status == 'Novi',
+          statusURadu: workOrder.status == 'U radu',
+          statusZatvoren: workOrder.status == 'Zatvoren'
+        }"
+        >{{ workOrder.status }}</span
+      >
     </div>
     <div class="col">
-      <div class="btn-group-vertical" style="float: right;">
-        <button class="btn btn-primary btn-edit mb-2" [disabled]="workOrder.status == 'Zatvoren'" *ngIf="!new"
-          (click)="closeWorkOrder()"><i class="fa fa-window-close mr-2"></i>Zatvori</button>
-        <button class="btn btn-primary btn-add mr-2" [disabled]="workOrder.status == 'Zatvoren'" *ngIf="!new"
-          (click)="saveData()"><i class="fa fa-save mr-2"></i>Sačuvaj</button>
-        <button class="btn btn-primary btn-add mr-2" *ngIf="new" (click)="addWorkOrder()"><i
-            class="fa fa-plus mr-2"></i>Dodaj</button>
+      <div class="btn-group-vertical" style="float: right">
+        <button
+          class="btn btn-primary btn-edit mb-2"
+          [disabled]="workOrder.status == 'Zatvoren'"
+          *ngIf="!new"
+          (click)="closeWorkOrder()"
+        >
+          <i class="fa fa-window-close mr-2"></i>Zatvori
+        </button>
+        <button
+          class="btn btn-primary btn-add mr-2"
+          [disabled]="workOrder.status == 'Zatvoren'"
+          *ngIf="!new"
+          (click)="updateWorkOrder()"
+        >
+          <i class="fa fa-save mr-2"></i>Sačuvaj
+        </button>
+        <button
+          class="btn btn-primary btn-add mr-2"
+          *ngIf="new"
+          (click)="addWorkOrder(workOrderInfoMobForm.valid)"
+        >
+          <i class="fa fa-plus mr-2"></i>Kreiraj
```

Comment:
Clicking this button sets red validation on chosen dates for a second, then routes to workOrder and displays a confirmation message even though the backend throws an exception, and work order was not created.